cluster/state.go

```diff
@@ -73,6 +74,21 @@ func (c *Cluster) GetClusterState(ctx context.Context) (*Cluster, error) {
 			if err != nil {
 				return nil, fmt.Errorf("Failed to Get Kubernetes certificates: %v", err)
 			}
+
+			// make sure I have all the etcd certs, We need handle dialer failure for etcd nodes https://github.com/rancher/rancher/issues/12898
+			for _, host := range activeEtcdHosts {
+				certName := pki.GetEtcdCrtName(host.InternalAddress)
+				if (currentCluster.Certificates[certName] == pki.CertificatePKI{}) {
```

Comment:
why using different condition than the one above

Reply:
This one is a completely empty object. The one above is passed through `populateCertMap()` which adds some static values like env name and file path.